src/App.jsx

```diff
@@ -67,13 +68,13 @@ class App extends React.Component {
   render() {
     let filteredHorns = this.state.data;
     if(this.state.filterBy === "1") {
-      filteredHorns = this.state.data.filter((horns) => horns[0] === "1");
+      filteredHorns = this.state.data.filter((beast) => beast.horns[0] === 1);
     } else if(this.state.filterBy === "2") {
-      filteredHorns = this.state.data.filter((horns) => horns[0] === "2");
+      filteredHorns = this.state.data.filter((beast) => beast.horns[0] === 2);
     } else if(this.state.filterBy === "other") {
-      filteredHorns = this.state.data.filter((horns) => horns[0] !== "1" && title[0] !== "2");
+      filteredHorns = this.state.data.filter((beast) => beast.horns[0] !== 1 && beast.horns[0] !== 2);
```

Comment:
beast.horns is not an array, just an integer! So just removing the [0] after each .horns and it should work!